.github/workflows/pr_validation.yml

```diff
@@ -0,0 +1,73 @@
+# ------------------------------------------------------------------------------
+# <auto-generated>
+#
+#     This code was generated.
+#
+#     - To turn off auto-generation set:
+#
+#         [CustomGitHubActions (AutoGenerate = false)]
+#
+#     - To trigger manual generation invoke:
+#
+#         nuke --generate-configuration GitHubActions_pr_validation --host GitHubActions
+#
+# </auto-generated>
+# ------------------------------------------------------------------------------
+
+name: pr_validation
+
+on:
+  push:
+    branches:
+      - master
+      - dev
+  pull_request:
+    branches:
+      - master
+      - dev
+
+jobs:
+  windows-latest:
+    name: windows-latest
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 5.0.*
+      - uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 6.0.*
+      - name: Cache .nuke/temp, ~/.nuget/packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            .nuke/temp
+            ~/.nuget/packages
+          key: ${{ runner.os }}-${{ hashFiles('global.json', 'src/**/*.csproj') }}
+      - name: Run './build.cmd RunTests'
```

Comment:
Why isn't `build.cmd all` a thing anymore? That allows us to also test that NuGet packaging, Docker builds, et al get tested on each project. Unit tests aren't the only thing that needs to run in order for a project to be safe.

The other important part of the "all" stage - stages that don't have the required inputs (i.e. `docker push`) skip themselves when those credentials are missing, rather than blow up.

Does that feedback make sense? This needs to be made more user-friendly.

Reply:
Also, `./build.cmd RunTests` fails on its own so this CI is broken at the moment. All stages should pull in their dependencies, like FAKE does.

Reply:
> Also, `./build.cmd RunTests` fails on its own so this CI is broken at the moment. All stages should pull in their dependencies, like FAKE does.

This should run, did you install nuke build tool? You can run `build.cmd install` to install

Reply:
The tool installed itself. I should be able to run everything in one shot without having to execute a separate command first. In this case it looks like the project failed because it wasn't built yet.